matrix.py

```diff
@@ -41,17 +41,58 @@ def mul(self, B):
         else:
             raise Exception("Illegal operation: incompatible dimensions")
 
-    def scalar(self,k):
+    def scalar(self, k):
         R = Matrix()
         R.r = self.r
         R.c = self.c
 
         for i in range(self.r):
             R.matrix.append(list([0] * self.c))
-            
 
         for i in range(self.r):
             for j in range(self.c):
-                R.matrix[i][j] = self.matrix[i][j]*k
+                R.matrix[i][j] = self.matrix[i][j] * k
 
-        return R
\ No newline at end of file
+        return R
+
+    def det(self):
+        matrix = self.matrix
+        return (
+            matrix[0][0] * (matrix[1][1] * matrix[2][2] - matrix[1][2] * matrix[2][1])
+            - matrix[0][1] * (matrix[1][0] * matrix[2][2] - matrix[1][2] * matrix[2][0])
+            + matrix[0][2] * (matrix[1][0] * matrix[2][1] - matrix[1][1] * matrix[2][0])
+        )
+
+    def adj(self):
+        matrix = self.matrix
+        adj = [[0, 0, 0] for _ in range(3)]
+
+        adj[0][0] = matrix[1][1] * matrix[2][2] - matrix[1][2] * matrix[2][1]
+        adj[0][1] = -(matrix[1][0] * matrix[2][2] - matrix[1][2] * matrix[2][0])
+        adj[0][2] = matrix[1][0] * matrix[2][1] - matrix[1][1] * matrix[2][0]
+
+        adj[1][0] = -(matrix[0][1] * matrix[2][2] - matrix[0][2] * matrix[2][1])
+        adj[1][1] = matrix[0][0] * matrix[2][2] - matrix[0][2] * matrix[2][0]
+        adj[1][2] = -(matrix[0][0] * matrix[2][1] - matrix[0][1] * matrix[2][0])
+
+        adj[2][0] = matrix[0][1] * matrix[1][2] - matrix[0][2] * matrix[1][1]
+        adj[2][1] = -(matrix[0][0] * matrix[1][2] - matrix[0][2] * matrix[1][0])
+        adj[2][2] = matrix[0][0] * matrix[1][1] - matrix[0][1] * matrix[1][0]
+
+        return adj
+
+    def inv(self):
+        matrix = self.matrix
+        det = self.det()
+
+        # Si el determinante es 0, la matrix no tiene inversa
```

Comment:
Spanish!!

Reply:
I am worry about the matrix transposed, try make the unitary prube, however the rest appear to be ok